### PR TITLE
feat: Added setLlmTokenCountCallback API endpoint to register a callback for calculating token count when none is provided

### LINKED ISSUE #2055: Add new api endpoint `setLlmTokenCountCallback` to provide ability to do token counting
### Description
If a customer chooses not to record LLM event ouput and input content then LLM token's cannot be attached to LLM message events server side. This means tokens must be calculated client side via registering a callback with `api.setLlmTokenCountCallback`.  In the openai and langchain instrumentation this callback will be invoke if it exists when token counts are not available. This callback will be invoked synchronously with the model and the message content/prompt and must return a token count which will be assigned to the relevant `token_count` attribute for the Llm event

### Acceptance Criteria
As a customer, I want to instrument my LLM calls, but I do not want to send the content to New Relic, I am willing to add additional instrumentation so that I can access all other aspects of the AI Monitoring UI

### PR DESCRIPTION
**Pull request by bizob2828**
## Description
Adds `setLlmTokenCountCallback` to register callback to be called to get token counts when they are not present on events.  This function must be synchronous and return token count.  

## Related Issues
Closes #2055

### CODE REVIEW

File: api.js [hunk 1]
```diff
@@ -1849,4 +1849,37 @@ API.prototype.setErrorGroupCallback = function setErrorGroupCallback(callback) {
...
+ * available. This function will typically only be used if `ai_monitoring.record_content.enabled` is false
+ * and you want to still capture token counts for Llm events.
+ *
+ * Provided callbacks must return an integer value for the token count for a given piece of content.
+ *
+ * @param {Function} callback - synchronous function called to calculate token count for content.
+ * @example
+ * // @param {string} model - name of model (i.e. gpt-3.5-turbo)
+ * // @param {string} content - prompt or completion response
+ * function tokenCallback(model, content) {
```
Comment (jsumners-nr) [inline api.js:1863]: The value of `content` is extremely varied. I can see us fielding plenty of issues asking what it will be.

**Review by jsumners-nr — APPROVED** [review]
Looks good to me.

File: api.js (hunk 1 above)
Comment (bizob2828) [reply api.js:1863]: It must be a string. In the openai cases they are always strings.  I know for langchain they aren't always strings but we don't assign tokens. Looking at bedrock they are strings as well

Comment (jsumners-nr) [reply api.js:1863]: https://github.com/jsumners-nr/node-newrelic-aws-sdk/blob/2b36ae2196983997177e1db7843120ac144d7fa7/tests/versioned/v3/bedrock-chat-completions.tap.js#L614-L627

https://github.com/jsumners-nr/node-newrelic-aws-sdk/blob/2b36ae2196983997177e1db7843120ac144d7fa7/lib/llm/bedrock-response.js#L170-L172

Comment (bizob2828) [reply api.js:1863]: it shouldn't be the parsed body. it should be the key in the body that contains the content.

**Pull request merged**